test/testdata/fixtures/client-encryption-secret.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: client-encryption-stores
```

Comment:
It's interesting to note that in `EncryptionStores` you distinguished 4 secret refs to define the keystore, its password, the truststore and its password. Yet here all 4 items are combined in one single secret, and hence the k8c definition references this single secret several times:

```
    server_encryption_stores:
      keystore_secret_ref:
        name: server-encryption-stores
      truststore_secret_ref:
        name: server-encryption-stores
      keystore_password_secret_ref:
        name: server-encryption-stores
      truststore_password_secret_ref:
        name: server-encryption-stores
    client_encryption_stores:
      keystore_secret_ref:
        name: client-encryption-stores
      truststore_secret_ref:
        name: client-encryption-stores
      keystore_password_secret_ref:
        name: client-encryption-stores
      truststore_password_secret_ref:
        name: client-encryption-stores
```

This hints at the fact that the granularity level is not optimal. I wonder if just one secret ref for keystore + truststore + passwords is enough? Or maybe two, one for the keystore, the other for the truststore?

Reply:
I simplified this to only use 2 secrets, containing both the store and the associated password.